src/hotspot/share/opto/convertnode.cpp

```diff
@@ -79,6 +79,11 @@ Node* Conv2BNode::Ideal(PhaseGVN* phase, bool can_reshape) {
         assert(false, "Unrecognized comparison for Conv2B: %s", NodeClassNames[in(1)->Opcode()]);
       }
 
+      // Skip the transformation if input is unexpected.
+      if (cmp == nullptr) {
+        return nullptr;
+      }
+
```

Comment:
@JohnTortugo Would it not have been better to put this check inside the `else` branch?

Reply:
Yes, putting `return nullptr;` into existing branch would have been cleaner. One more reason to wait for reviews! We can do a quick follow-up that sweeps this return to its better place, if you feel strongly about it. It is about 15 minute deal for me.